tensorflow_probability/python/sts/dynamic_regression.py

```diff
@@ -0,0 +1,172 @@
+from tensorflow_probability.python.sts.internal import util as sts_util
+from tensorflow_probability.python.sts.structural_time_series import Parameter
+from tensorflow_probability.python.sts.structural_time_series import StructuralTimeSeries
+
+
+class DynamicLinearRegressionStateSpaceModel(tfd.LinearGaussianStateSpaceModel):
+
+  def __init__(self,
+               num_timesteps,
+               design_matrix,
+               weights_scale,
+               initial_state_prior,
+               observation_noise_scale=0.,
+               initial_step=0,
+               validate_args=False,
+               allow_nan_stats=True,
+               name=None):
+
+    with tf.compat.v1.name_scope(name, 'DynamicLinearRegressionStateSpaceModel',
+                                 values=[weights_scale]) as name:
+
+      dtype = dtype_util.common_dtype(
+          [design_matrix, weights_scale, initial_state_prior])
+
+      design_matrix = tf.convert_to_tensor(
+          value=design_matrix, name='design_matrix', dtype=dtype)
+
+      weights_scale = tf.convert_to_tensor(
+          value=weights_scale, name='weights_scale', dtype=dtype)
+
+      observation_noise_scale = tf.convert_to_tensor(
+          value=observation_noise_scale,
+          name='observation_noise_scale',
+          dtype=dtype)
+
+      num_features = tf.shape(design_matrix)[-1]
+
+      def observation_matrix_fn(t):
+        observation_matrix = tf.linalg.LinearOperatorFullMatrix(
+            design_matrix[..., t, tf.newaxis, :], name='observation_matrix')
+        return observation_matrix
+
+      self._weights_scale = weights_scale
+      self._observation_noise_scale = observation_noise_scale
+
+      super(DynamicLinearRegressionStateSpaceModel, self).__init__(
+          num_timesteps=num_timesteps,
+          transition_matrix=tf.linalg.LinearOperatorIdentity(
+              num_rows=num_features,
+              dtype=dtype,
+              name='transition_matrix'),
+          transition_noise=tfd.MultivariateNormalDiag(
+              scale_diag=tf.broadcast_to(weights_scale, [num_features]),
+              name='transition_noise'),
+          observation_matrix=observation_matrix_fn,
+          observation_noise=tfd.MultivariateNormalDiag(
+              scale_diag=observation_noise_scale[..., tf.newaxis],
+              name='observation_noise'),
+          initial_state_prior=initial_state_prior,
+          initial_step=initial_step,
+          allow_nan_stats=allow_nan_stats,
+          validate_args=validate_args,
+          name=name)
+
+  @property
+  def weights_scale(self):
+    """Standard deviation of the weights transitions."""
+    return self._weights_scale
+
+  @property
+  def observation_noise_scale(self):
+    """Standard deviation of the observation noise."""
+    return self._observation_noise_scale
+
+
+class DynamicLinearRegression(StructuralTimeSeries):
+
+  def __init__(self,
+               design_matrix,
+               weights_scale_prior=None,
```

Comment:
Yes I agree having `drift` in the name is a lot clearer. I think both the names you suggested are good - I went with the former in 242b0fff826eee7145e8a81f7deda3674c1ef29a since it is shorter but would be happy to rename again.

Reply:
sgtm. thanks!

Reply:
pls ignore if not interesting, but it might also be worth considering renaming this in some of the other components, for example in local level you have:

```python
level[t] = level[t-1] + Normal(0., level_scale)
```

Should this be `level_drift`?